bob-common/src/node.rs

```diff
@@ -72,16 +74,22 @@ impl Node {
 
     pub async fn set_connection(&self, client: BobClient) {
         *self.conn.write().await = Some(client);
+        self.conn_available.store(true, Ordering::Relaxed);
```

Comment:
Do this update inside `self.conn.write()` lock. Otherwise the following situation is possible:
1. Thread1: `set_connection` is called and update connection
2. Thread2: `clear_connection` is called and reset connection
3. Thread2: store `false` to `conn_available`
4. Thread1: store `true` to `conn_available`